subprojects/core-api/src/main/java/org/gradle/api/DomainObjectCollection.java

```diff
@@ -168,6 +192,8 @@ public interface DomainObjectCollection<T> extends Collection<T> {
     // note: this is here to override the default Groovy Collection.findAll { } method.
     /**
      * Returns a collection which contains the objects in this collection which meet the given closure specification.
+     * <p>
+     * This method is an intermediate eager operation. The application of the filter forces the realization of the collection elements.
```

Comment:
Is it intermediate? It doesn't return a `DomainObjectCollection`, just a simple `Collection`, so you can't chain it further.

Reply:
Good point, updating!